lib/job.js

```diff
@@ -232,25 +233,84 @@ class Job extends BaseModel {
      * @return {Promise}  Resolves to array of metrics for builds belong to this job
      */
     async getMetrics(config = { startTime: null, endTime: null }) {
-        // Get builds during this time range
-        const builds = await this.getBuilds({
+        const options = {
             startTime: config.startTime,
             endTime: config.endTime,
             sort: 'ascending',
             paginate: {
                 count: MAX_COUNT
             }
-        });
-
-        // Generate metrics
-        const metrics = builds.map((b) => {
-            const { id, jobId, eventId, createTime, status, startTime, endTime } = b;
-            const duration = Math.round((new Date(endTime) - new Date(startTime)) / 1000);
+        };
+        const findMetrics = ({ id, jobId, eventId, createTime, status, startTime, endTime }) => {
+            const duration = startTime && endTime
+                ? dayjs(endTime).diff(dayjs(startTime), 'second')
+                : null;
 
             return { id, jobId, eventId, createTime, status, duration };
+        };
+
+        if (!config.aggregate) {
+            // Get builds during this time range
+            const builds = await this.getBuilds(options);
+
+            // Generate metrics
+            return builds.map(b => findMetrics(b));
+        }
+
+        const formatDate = dateTime => dayjs(dateTime).format('YYYY-MM-DD');
+
+        // recursively fetching builds until the end
+        const getAllBuilds = async (opts, buildArray, date, index) => {
+            const builds = await this.getBuilds(opts);
+            const length = builds.length;
+
+            if (length === 0) {
+                return buildArray;
+            }
+
+            let currentDate = date || formatDate(builds[0].createTime);
+            let currentIndex = index || 0;
+
+            // Create an array where each element is an array of builds of the same date
+            for (let i = 0; i < length; i += 1) {
+                const currentBuild = builds[i];
+                const buildDate = formatDate(currentBuild.createTime);
+
+                if (!dayjs(buildDate).isSame(dayjs(currentDate))) {
+                    currentIndex += 1;
+                    currentDate = buildDate;
+                    buildArray.push([]);
+                }
+
+                buildArray[currentIndex].push(currentBuild);
```

Comment:
Not sure I understand here. If the buildDate is the same as current date, the subsequent builds are pushed to the same index every time?

Reply:
If buildDate is same as currentDate then push to the same index. If it's different, then increment the index & set currentDate to the new buildDate. 

It will be an array of array 
```
[[all builds on 3/22],[all builds on date 3/23],[all builds on 3/24]...]
```
so that we can compute the daily avg more easily later.